### PR TITLE
Replace flatpickr with Reka-ui components

### LINKED ISSUE #25341: Replace flatpickr
### Describe the Bug

The flatpickr library used in the `<v-date-picker>` component is outdated, and should be replaced with a library that offers the same or better features. It should support a11y.

### To Reproduce

https://github.com/flatpickr/flatpickr/discussions/2479

### Directus Version

v11.8.0

### Hosting Strategy

Self-Hosted (Docker Image)

### Database

_No response_

### PR DESCRIPTION
**Pull request by alvarosabu**
## Scope

What's changed:

- Replaced flatpickr with Reka UI's [Calendar](https://reka-ui.com/docs/components/calendar#calendar) and [TimeField](https://reka-ui.com/docs/components/time-field) components. 
- Added `@internationalized/date` as recommended by Reka to operate with it's components
- Refactor `v-date-picker.vue` to use Reka UI primitives
- Added custom calendar heading inputs (Month and Year)
- Added custom "Set to now" button
- Created new utility `format-date-picker-model-value.ts` to handle date formatting logic
- Removed flatpickr dependency and associated files (`get-flatpickr-locale.ts`, `_flatpickr.scss`)
- Improved RTL support using native Reka UI `dir` prop integration with user text direction preference

## Benefits
- Reka UI was already a dependency, so bundle size trade-off of `@internationalized/date`(~15KB) is small vs flatpickr (~17KB)
- No breaking changes on the public API as far as my tests go.

## Potential Risks / Drawbacks

- Some edge cases due the complete rewrite
- There are some visual differences
- The custom implementation of the calendar heading inputs (Month and Year) might lead to visual issues with `data-` attributes like day selected.

## Tested Scenarios

- Date-only picker mode (`type="date"`)
- Time-only picker mode (`type="time"`)
- DateTime combined picker mode (`type="dateTime"`)
- Timestamp mode (`type="timestamp"`)
- 12-hour and 24-hour format toggle (`use24` prop)
- Seconds display toggle (`includeSeconds` prop)
- "Set to Now" button functionality
- Disabled state
- RTL text direction (using user store text direction preference)
- Integration with datetime interface
- Integration with system filter input component
- Model value formatting and parsing for all picker types
- Tested on Safari and Mobile

## Review Notes / Questions

- The component now uses `@internationalized/date` types (CalendarDate, Time, CalendarDateTime, ZonedDateTime) internally but maintains ISO string format for the public API
- Keyboard navigation and accessibility should be thoroughly tested as Reka UI provides enhanced a11y features that weren't present in flatpickr

## Checklist
- [x] Add localization for "Set to now" button
- [x] Added or updated tests
- [x] Documentation PR created [here](https://github.com/directus/docs/pull/549) or not required 
- ~~[ ] OpenAPI package PR created [here](https://github.com/directus/openapi) or not required~~

---

Fixes #25341

### CODE REVIEW

**Comment by AlexGaillard** [comment]
@HZooly is gonna give this a look as well 😄

File: app/src/utils/format-date-picker-model-value.ts [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+): string | null {
+	if (type === 'date') {
+		if (!calendarValue) return null;
+		return calendarValue.toString(); // "yyyy-MM-dd"
+	}
+
+	if (type === 'time') {
+		if (!timeValue) return null;
+		// Legacy emits `HH:mm:ss` even when seconds UI is hidden (seconds become "00").
+		const seconds = includeSeconds ? (timeValue.second ?? 0) : 0;
```
Comment (HZooly) [inline app/src/utils/format-date-picker-model-value.ts]: is nullish coaleshing needed here? `timeValue.second` is typed `number` only.

File: app/src/components/v-date-picker/v-date-picker.vue [hunk 2]
```diff
@@ -0,0 +1,564 @@
...
+});
+
+watch(
+	() => props.modelValue,
+	(newValue) => {
+		if (!newValue) {
+			calendarValue.value = undefined;
+
+			// Reset time to a sensible default when the picker supports time.
+			internalTimeValue.value = ['time', 'dateTime', 'timestamp'].includes(props.type)
```
Comment (HZooly) [inline app/src/components/v-date-picker/v-date-picker.vue]: This check occurs 3 times in the component, can we extract it in a computed?
```ts
const hasTime = computed(() => ['time', 'dateTime', 'timestamp'].includes(props.type));
```

File: app/src/components/v-date-picker/v-date-picker.vue [hunk 3]
```diff
@@ -0,0 +1,564 @@
...
+}
+
+const props = withDefaults(defineProps<Props>(), {
+	modelValue: undefined,
+	disabled: false,
+	includeSeconds: false,
+	use24: true,
+});
+
+const emit = defineEmits<{ 'update:modelValue': [value: string | null]; close: [] }>();
```
Comment (HZooly) [inline app/src/components/v-date-picker/v-date-picker.vue]: `close` is not used in this component, is it somewhere else?

File: app/src/components/v-date-picker/v-date-picker.vue [hunk 4]
```diff
@@ -0,0 +1,564 @@
...
+	<div class="v-date-picker" tabindex="-1">
+		<CalendarRoot
+			v-slot="{ weekDays, grid, date }"
+			v-model="calendarValue"
+			:disabled="disabled"
+			class="calendar"
+			fixed-weeks
+			weekday-format="short"
+			:dir="isRTL ? 'rtl' : 'ltr'"
+			@update:model-value="handleDateChange"
```
Comment (HZooly) [inline app/src/components/v-date-picker/v-date-picker.vue:256]: I think having `v-model` with `@update:model-value` can lead to a double update of the value, maybe it's enough to have 

```html
:model-value="calendarValue"
@update:model-value="handleDateChange"
```

File: app/src/components/v-date-picker/v-date-picker.vue [hunk 5]
```diff
@@ -0,0 +1,564 @@
...
+	}
+
+	.calendar-cell-trigger:hover {
+		color: var(--theme--foreground);
+		background-color: var(--background-highlight);
+		border-color: var(--background-highlight);
+	}
+
+	.calendar-cell-trigger[data-disabled] {
+		cursor: none;
```
Comment (HZooly) [inline app/src/components/v-date-picker/v-date-picker.vue]: Shouldn't be `not-allowed` here? 
```suggestion
		cursor: not-allowed;
```

File: app/src/components/v-date-picker/v-date-picker.vue [hunk 6]
```diff
@@ -0,0 +1,564 @@
...
+	if (showTime.value) {
+		internalTimeValue.value = new Time(now.getHours(), now.getMinutes(), now.getSeconds());
+	}
+
+	emitValue();
+}
+</script>
+
+<template>
+	<div class="v-date-picker" tabindex="-1">
```
Comment (HZooly) [inline app/src/components/v-date-picker/v-date-picker.vue]: Is the negative tabIndex important here? I don't see any difference without

**Review by HZooly — COMMENTED** [review]
Overall very good, thanks for the well commented `v-date-picker` component!
Some details/questions, feel free to ignore if not relevant.

**Comment by HZooly** [comment]
I'm not sure that having a search input for months - which is a static array - is relevant?
CleanShot 2026-02-05 at 13 18 01@2x

**Comment by alvarosabu** [comment]
> I'm not sure that having a search input for months - which is a static array - is relevant? CleanShot 2026-02-05 at 13 18 01@2x

@HZooly  I added it because it existed in the previous library but not supported by reka, otherwise would be a breaking change.

File: pnpm-workspace.yaml [hunk 7]
```diff
@@ -267,7 +268,7 @@ catalog:
   qrcode: 1.5.4
   qs: 6.14.1
   rate-limiter-flexible: 7.2.0
-  reka-ui: 2.6.0
+  reka-ui: 2.7.0
```
Comment (HZooly) [inline pnpm-workspace.yaml]: Reka UI released `v2.8.0` [last week](https://github.com/unovue/reka-ui/releases/tag/v2.8.0), with some breaking changes on time/date components. Do you want to upgrade now in this pull request, or keep in mind and do it later?

Comment (alvarosabu) [reply pnpm-workspace.yaml]: @formfcw since is not prio maybe is better I update the dependency and solve any issue first, wydt?

Comment (alvarosabu) [reply pnpm-workspace.yaml]: @HZooly seems like the breaking change shouldnt affect the impl 

DatePicker/DateRangePicker/Calendar/RangeCalendar: Make weekStartsOn locale-independent  -  by @kricsleo in https://github.com/unovue/reka-ui/pull/2359 [(623df)](https://github.com/unovue/reka-ui/commit/623dff46) 

But I'm testing either way

**Review by HZooly — APPROVED** [review]
LGTM, thanks for the changes!

File: pnpm-workspace.yaml (hunk 7 above)
Comment (alvarosabu) [reply pnpm-workspace.yaml]: Confirm, no issues founded

File: app/src/lang/translations/en-US.yaml [hunk 8]
```diff
@@ -2904,3 +2904,4 @@ accepted: Accepted
 text_direction_auto: Automatic (Match Language)
 text_direction_ltr: Left to Right (LTR)
 text_direction_rtl: Right to Left (RTL)
+datetime_picker_now_btn_label: Set To Now
```
Comment (rijkvanzanten) [inline app/src/lang/translations/en-US.yaml]:
```suggestion
set_to_now: Set to Now
```

Comment (rijkvanzanten) [reply app/src/lang/translations/en-US.yaml]: That way it can be reused elsewhere

File: app/src/components/v-date-picker/v-date-picker.vue [hunk 9]
```diff
@@ -0,0 +1,570 @@
...
+							</TimeFieldInput>
+							<TimeFieldInput v-else :part="item.part" class="time-field-segment">
+								{{ item.value }}
+							</TimeFieldInput>
+						</template>
+					</TimeFieldRoot>
+				</div>
+				<!-- Today Button -->
+				<div class="calendar-footer">
+					<button class="calendar-today-button" @click="setToNow">{{ $t('datetime_picker_now_btn_label') }}</button>
```
Comment (rijkvanzanten) [inline app/src/components/v-date-picker/v-date-picker.vue]:
```suggestion
					<button class="calendar-today-button" @click="setToNow">{{ $t('set_to_now') }}</button>
```

**Review by rijkvanzanten — APPROVED** [review]
LGTM!
